### DISCUSSION TITLE
Implement Klipper Estimator for Second acurate time estimates for Klipper FW printers.

### DISCUSSION THREAD

**Discussion by TheBest6337** [OrcaSlicer/OrcaSlicer#2009] [Ideas]
Is your feature request related to a problem? Please describe.
If you are using a 3d printer with the Klipper firmware you get semi Accurate time predictions, that can be way too short or way to long.

Which printers will be beneficial to this feature
Every printer that uses the Klipper firmware

Describe the solution you'd like
With the Klipper Estimator post-processing script, it is much closer (About +/-5 seconds) to the actual time. You can use the estimator as a post processor, but that comes with the drawback that you only get the estimate after processing (which happens after uploading or saving the gcode) so you don't get the estimate shown in the Slicer and only in your Klipper interface. I would like that this gets implemented so that it's directly in the Orca shown estimate if a printer with Klipper is connected via Mainsail.

Describe alternatives you've considered
I've tried to get more accurate estimates by defining the maximum Acceleration, but the estimates also were not near the actual time.

Additional context
A link to the Klipper Estimator is here: https://github.com/Annex-Engineering/klipper_estimator

**Comment by agravelot** [comment]
Linked to: https://github.com/SoftFever/OrcaSlicer/issues/1612
